### PR TITLE
Add cloud teams integration

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Add gRPC contract. I used `oneof` approach, similar to https://github.com/grpc/grpc-proto/blob/154a1b24afac83e28240df68b48ebb063d66ed4b/grpc/lb/v1/load_balancer.proto#L36
- Add cloud teams platform to process incoming events from cloud side and also send source events to cloud

## Related issue(s)

- https://github.com/kubeshop/botkube-cloud/issues/670

### CODE REVIEW

File: pkg/bot/teams_cloud.go [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+func (b *CloudTeams) GetStatus() health.PlatformStatus {
+	return health.PlatformStatus{
+		Status:   b.status,
+		Restarts: fmt.Sprintf("%d/%d", b.failuresNo, maxRetries),
+		Reason:   b.failureReason,
+	}
+}
+
+func (b *CloudTeams) start(ctx context.Context) error {
+	svc, err := newGrpcCloudTeamsConnector(b.log, b.cfg.Server.URL)
```
Comment (huseyinbabal) [inline pkg/bot/teams_cloud.go:182]: better to add error context to separate it from next one

File: pkg/bot/teams_cloud.go [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+		channel, exists := b.getChannelForActivity(act)
+
+		msg := b.processMessage(ctx, act, channel, exists)
+		if msg.IsEmpty() {
+			return nil, nil
+		}
+
+		raw, err := json.Marshal(msg.Message)
+		if err != nil {
+			return nil, err
```
Comment (huseyinbabal) [inline pkg/bot/teams_cloud.go:268]: error context?

File: pkg/bot/teams_cloud_grpc.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+func (c *grpcCloudTeamsConnector) Start(ctx context.Context) error {
+	activityClient, err := pb.NewCloudTeamsClient(c.grpcConn).StreamActivity(ctx)
+	if err != nil {
+		return fmt.Errorf("while initializing gRPC cloud client: %w", err)
+	}
+
+	err = activityClient.Send(&pb.AgentActivity{
+		Req: &pb.AgentActivity_InstanceId{
+			//InstanceId: c.remoteConfig.Identifier, // TODO: enable when config generator will work
+			InstanceId: "foo-123",
```
Comment (huseyinbabal) [inline pkg/bot/teams_cloud_grpc.go:85]: So, `foo-123` will be replaced by a dynamic logic?

**Review by huseyinbabal — APPROVED** [review]
:rocket: with minor comments

File: pkg/bot/teams_cloud_grpc.go (hunk 3 above)
Comment (mszostok) [reply pkg/bot/teams_cloud_grpc.go:85]: yes, but it will be done once the config generator will be in place 👍

File: pkg/bot/teams_cloud.go (hunk 2 above)
Comment (mszostok) [reply pkg/bot/teams_cloud.go:268]: fixed

**Pull request merged**